src/core/services/compiler/compiler.spec.js

```diff
@@ -159,31 +166,71 @@ describe('$mdCompiler service', function() {
         expect(scope.myControllerAs).toBe(data.element.controller());
       }));
 
-      it('should work with bindToController', inject(function($rootScope) {
-        var called = false;
-        var data = compile({
-          template: 'hello',
-          controller: TestController,
-          controllerAs: 'ctrl',
-          bindToController: true,
-          locals: { name: 'Bob' }
-        });
+    });
 
-        var scope = $rootScope.$new();
-        data.link(scope);
-        expect(scope.ctrl.name).toBe('Bob');
-        expect(called).toBe(true);
+  });
 
+  describe('with preAssignBindings', function() {
 
-        function TestController($scope) {
-          called = true;
-          expect($scope.$apply).toBeTruthy(); // Ensure that DI works properly
-        }
+    function compileAndLink(options) {
+      var data = compile(options);
 
-        TestController.prototype.$onInit = function() {
-          expect(this.name).toBe('Bob');
-        };
-      }));
+      inject(function($rootScope) {
+        data.link($rootScope);
+      });
+
+      return data;
+    }
+
+    it('enabled should assigns bindings at instantiation', function() {
+      configureCompiler(true);
+
+      var isInstantiated = false;
+
+      function TestController($scope) {
+        isInstantiated = true;
+        expect($scope.$apply).toBeTruthy();
+        expect(this.name).toBe('Bob');
+      }
+
+      compileAndLink({
+        template: 'hello',
+        controller: TestController,
+        controllerAs: 'ctrl',
+        bindToController: true,
+        locals: { name: 'Bob' }
+      });
+
+      expect(isInstantiated).toBe(true);
+    });
+
+    it('disabled should assigns bindings after constructor', function() {
+      configureCompiler(false);
```

Comment:
What advantage does this have? I think the spec name and function parameter already explains that this changes the `preAssignBindings` behavior.